modules/base/base/module.py

```diff
@@ -307,10 +307,18 @@ async def autothread_list(self, ctx):
 
     @commands.check(check.acl)
     @autothread.command(name="set")
-    async def autothread_set(self, ctx, channel: discord.TextChannel = None):
-        if channel is None:
-            channel = ctx.channel
-        AutoThread.add(ctx.guild.id, channel.id)
+    async def autothread_set(self, ctx, channel: discord.TextChannel, duration: str):
+        durations = {"1h": 60, "1d": 1440, "3d": 4320, "7d": 10080}
+        try:
+            duration_translated = durations[duration]
+        except KeyError:
+            await ctx.reply(tr("autothread set", "duration", ctx))
+            return
+        if ctx.guild.premium_tier < 2 and duration_translated == durations["7d"]:
+            duration_translated = durations["3d"]
+        if ctx.guild.premium_tier < 1 and duration_translated == durations["3d"]:
+            duration_translated = durations["1d"]
```

Comment:
I think we should allow them to set the value over the current maximum, because the guild level may change over time. I'd guard the maximum value at the time of thread creation, not here.